src/client/pythonEnvironments/legacyIOC.ts

```diff
@@ -204,30 +206,28 @@ class ComponentAdapter implements IComponentAdapter {
         if (!(await this.isEnabled())) {
             return undefined;
         }
-        const env = await this.api.resolveEnv(interpreterPath);
-        if (env === undefined) {
-            return undefined;
-        }
-        return env.kind === PythonEnvKind.Conda;
+        // Eventually we won't be calling 'isCondaEnvironment' in the component adapter, so we won't
+        // need to use 'isCondaEnvironment' directly here. This is just a temporary implementation.
+        return isCondaEnvironment(interpreterPath);
     }
 
     // A result of `undefined` means "Fall back to the old code!"
     public async getCondaEnvironment(interpreterPath: string): Promise<CondaEnvironmentInfo | undefined> {
         if (!(await this.isEnabled())) {
             return undefined;
         }
-        const env = await this.api.resolveEnv(interpreterPath);
-        if (env === undefined) {
-            return undefined;
-        }
-        if (env.kind !== PythonEnvKind.Conda) {
+        // For Conda we assume we don't set name for environments if they're prefix conda environments, similarly
+        // we don't have 'path' set if they're non-prefix conda environments.
+        // tslint:disable-next-line:no-suspicious-comment
+        // TODO: Expose these two properties via a method in the Conda locator on a temporary basis. This method
+        // will go away eventually, and so exposing those properties will no longer be needed. Until then we only
+        // return location.
+        const location = getEnvironmentDirFromPath(interpreterPath);
```

Comment:
```suggestion
        // While `ComponentAdapter` represents how the component would be used in the rest of the
        // extension, we cheat here for the sake of performance.  This is not a problem because when
        // we start using the component's public API directly we will be dealing with `PythonEnvInfo`
        // instead of just `pythonPath`.
        //
        // Also, for Conda we assume we don't set name for environments if they're prefix conda
        // environments, similarly we don't have 'path' set if they're non-prefix conda environments.
        // tslint:disable-next-line:no-suspicious-comment
        // TODO: Expose these two properties via a method in the Conda locator on a temporary basis. This method
        // will go away eventually, and so exposing those properties will no longer be needed. Until then we only
        // return location.
        const location = getEnvironmentDirFromPath(interpreterPath);
```

(Presumably this change was for the sake of performance...)